filebeat/input/file/state.go

```diff
@@ -49,5 +56,5 @@ func (s *State) IsEqual(c *State) bool {
 
 // IsEmpty returns true if the state is empty
 func (s *State) IsEmpty() bool {
-	return *s == State{}
+	return reflect.DeepEqual(*s, State{})
```

Comment:
we need `DeepEqual`?

Reply:
How about `tmp := *s; tmp.Meta = nil; return tmp == State{} && len(s.Meta) == 0`

Reply:
The compiler still says it cannot compare maps, are your concerns about performance? I could do manual comparison for all fields, or use 2 structs like:

```
type State struct {
   Common
   Meta        map[string]string `json:"meta"`
}
```

then:

```
if len(s.Meta) > 0 { return false }
return *s.Common == Common{}
```

Reply:
a little performance, but I'd also prefer to keep out `reflect` from most packages.

I wonder if checking for FileStateOS or the timestamp only would be enough. This method is used to determine if we have a valid/usable state or not.

Reply:
I don't have enough experience with all the possible situations with the registry so I'll trust you here on what to do :innocent:

Reply:
I updated this to use `s.FileStateOS != file.StateOS{} && !s.Timestamp.IsZero()`, also pinging @ruflin for awareness